shell/renderer/api/electron_api_context_bridge.cc

```diff
@@ -169,24 +126,27 @@ v8::MaybeLocal<v8::Value> PassValueToOtherContext(
   // the global handle at the right time.
   if (value->IsFunction()) {
     auto func = v8::Local<v8::Function>::Cast(value);
-    v8::Global<v8::Function> global_func(source_context->GetIsolate(), func);
-    v8::Global<v8::Context> global_source(source_context->GetIsolate(),
-                                          source_context);
 
-    size_t func_id = store->take_func_id();
-    store->functions()[func_id] =
-        std::make_tuple(std::move(global_func), std::move(global_source));
     {
       v8::Context::Scope destination_scope(destination_context);
-      v8::Local<v8::Value> proxy_func = gin_helper::CallbackToV8Leaked(
-          destination_context->GetIsolate(),
-          base::BindRepeating(&ProxyFunctionWrapper, store, func_id,
-                              support_dynamic_properties));
-      FunctionLifeMonitor::BindTo(destination_context->GetIsolate(),
-                                  v8::Local<v8::Object>::Cast(proxy_func),
-                                  store->GetWeakPtr(), func_id);
-      object_cache->CacheProxiedObject(value, proxy_func);
-      return v8::MaybeLocal<v8::Value>(proxy_func);
+      gin_helper::Dictionary state = gin_helper::Dictionary::CreateEmpty(
+          destination_context->GetIsolate());
+      DCHECK(IsTrue(state.GetHandle()->SetPrivate(
```

Comment:
anything inside `DCHECK(...)` will be compiled out in non-debug builds. This should be `CHECK`.